From 37625d6dd5f8b566a5cd74f308bf6bd8244c6037 Mon Sep 17 00:00:00 2001
From: Tao Ren <taoren@fb.com>
Date: Mon, 11 May 2020 13:51:25 -0700
Subject: [PATCH 132/132] ARM: dts: aspeed-g5: add jtag controller

Summary:
Add JTAG controller in aspeed-g5.dtsi. The patch is generated from
D21517642 and it's applied to different kernel version.

Test Plan:
1) "bitbake wedge400-image" succeeded for kernel 5.15.
2) verified jtag device is bound to the driver correctly on wedge400.
3) "scmcpld_update.sh" can be used to upgrade scm firmware on wedge400
     successfully.

Reviewers: tfang, goldenbug, #openbmc

Reviewed By: goldenbug

Differential Revision: https://phabricator.intern.facebook.com/D37292797
---
 arch/arm/boot/dts/aspeed/aspeed-g5.dtsi | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/arch/arm/boot/dts/aspeed/aspeed-g5.dtsi b/arch/arm/boot/dts/aspeed/aspeed-g5.dtsi
index 04f98d1dbb97..f4d6332d3bbd 100644
--- a/arch/arm/boot/dts/aspeed/aspeed-g5.dtsi
+++ b/arch/arm/boot/dts/aspeed/aspeed-g5.dtsi
@@ -270,6 +270,15 @@ hace: crypto@1e6e3000 {
 				resets = <&syscon ASPEED_RESET_HACE>;
 			};
 
+			jtag: jtag@1e6e4000 {
+				compatible = "aspeed,ast2500-jtag";
+				reg = <0x1e6e4000 0x1c 0x1e6e2004 0x04>;
+				clocks = <&syscon ASPEED_CLK_APB>;
+				resets = <&syscon ASPEED_RESET_JTAG_MASTER>;
+				interrupts = <43>;
+				status = "disabled";
+			};
+
 			gfx: display@1e6e6000 {
 				compatible = "aspeed,ast2500-gfx", "syscon";
 				reg = <0x1e6e6000 0x1000>;
-- 
2.43.0

